engine/src/org/pentaho/di/trans/steps/rest/Rest.java

```diff
@@ -202,11 +207,43 @@ public Rest( StepMeta stepMeta, StepDataInterface stepDataInterface, int copyNr,
 
       // Get Response
       String body;
+      StringBuilder headerString = new StringBuilder();
       try {
         body = response.getEntity( String.class );
       } catch ( UniformInterfaceException ex ) {
         body = "";
       }
+      
+      // get Header
+      MultivaluedMap<String, String> headers = response.getHeaders();
+      
+      headerString.append("{");
```

Comment:
As noted above. Please build JSON using an org.json.simple.JSONObject as shown in JsonOutput step (org.pentaho.di.trans.steps.jsonoutput).